packages/react/src/Delay.tsx

```diff
@@ -15,11 +25,20 @@ export const Delay = (props: DelayProps) => {
   const defaultProps = useContext(DelayDefaultPropsContext)
   const ms = props.ms ?? defaultProps.ms ?? 0
 
-  const [isDelaying, setIsDelaying] = useState(ms > 0)
-  useTimeout(() => setIsDelaying(false), ms)
+  const [isDelayed, setIsDelayed] = useState(ms <= 0)
+  useTimeout(() => setIsDelayed(true), ms)
+
+  if (typeof props.children === 'function') {
+    return <>{props.children({ isDelayed })}</>
+  }
 
-  const fallback = props.fallback === undefined ? defaultProps.fallback : props.fallback
-  return <>{isDelaying ? fallback : props.children}</>
+  if (isDelayed) {
+    return <>{props.children}</>
+  }
+  if (props.fallback === undefined) {
+    return <>{defaultProps.fallback}</>
+  }
```

Comment:
Is it correct behavior that when props.fallback is null, the fallback from defaultProps doesn't activate?

Reply:
That's right! I added test code to prove this as spec in [8960583](https://github.com/toss/suspensive/pull/1312/commits/896058394720a8bd770d1c2bd358db1ccf83b351) and I added you as co-author